### PR TITLE
Fail-fast on DDTracer.build when writer is null or DDAgentWriter

### PR DESCRIPTION
**Pull request by drodriguezhdez**
After #1994, it's not possible to use the `DDTracer` with the `ListWriter` for tests and using the `dd-java-agent` in the test phase to report tests using the `JUnit4` instrumentation.

This `PR` modifies the logic to fail-fast only when the writer is `null` or `DDAgentWriter`.

### CODE REVIEW

File: dd-trace-ot/src/main/java/datadog/opentracing/DDTracer.java [hunk 1]
```diff
@@ -200,10 +200,13 @@ private DDTracer(
...
-          "Datadog Tracer already installed by `dd-java-agent`. NOTE: Manually creating the tracer while using `dd-java-agent` is not supported");
-      throw new IllegalStateException("Datadog Tracer already installed");
+    if (writer == null
+        || writer.getClass().getName().equals("datadog.trace.common.writer.DDAgentWriter")) {
+      if (GlobalTracer.get().getClass().getName().equals("datadog.trace.agent.core.CoreTracer")) {
+        log.error(
+            "Datadog Tracer already installed by `dd-java-agent`. NOTE: Manually creating the tracer while using `dd-java-agent` is not supported");
+        throw new IllegalStateException("Datadog Tracer already installed");
+      }
     }
```
Comment (randomanderson) [inline dd-trace-ot/src/main/java/datadog/opentracing/DDTracer.java:210]:
```suggestion
    if ((writer == null
        || writer.getClass().getName().equals("datadog.trace.common.writer.DDAgentWriter")) &&
      GlobalTracer.get().getClass().getName().equals("datadog.trace.agent.core.CoreTracer")) {
        log.error(
            "Datadog Tracer already installed by `dd-java-agent`. NOTE: Manually creating the tracer while using `dd-java-agent` is not supported");
        throw new IllegalStateException("Datadog Tracer already installed");
    }
```

Comment (randomanderson) [reply dd-trace-ot/src/main/java/datadog/opentracing/DDTracer.java:210]: These if statements can be combined.  I probably messed up the indentation using the Github web editor

**Review by randomanderson — APPROVED** [review]
I'm okay with this change because it's harmless but I find it hard to come up with a situation where this helps.

The user would have to:
- Use the javaagent in the test
- Manually create `DDTracer` in the test
- Know enough to not use `DDAgentWriter`
- Not use `GlobalTracer.install()` on the manually created `DDTracer`
- Have code that doesn't rely on the global tracer being installed

File: dd-trace-ot/src/main/java/datadog/opentracing/DDTracer.java (hunk 1 above)
Comment (drodriguezhdez) [reply dd-trace-ot/src/main/java/datadog/opentracing/DDTracer.java]: Combined 👍

**Review by bantonsson — APPROVED** [review]

**Pull request merged**